### ISSUE TITLE
[PS Vita] Crackling sounds and no menu sounds

### ISSUE THREAD

**Issue by evillurker** [libretro/RetroArch#12543]
Device: PS Vita
Version: Latest stable, latest nightly
If you enter the menu and then resume game, it will start crackling sounds non stop until you restart the game

Seems like turning on threaded video solves this but it will cause a bunch of other issues:
When threaded video is on, DOSBox-SVN will crash any game you try to run through it.

Using the cores from March 10th nightly build solves this audio issue but if you use those cores you will not get menu sounds + menu BGM.

**Comment by markwkidd** [comment]
FWIW I will add that this seems to be affecting multiple cores even though the steps to reproduce are for dosbox-svn.

**Comment by twinaphex** [comment]
We need to know what exact commit from March 10 to today caused this.

**Comment by markwkidd** [comment]
@evillurker would you be willing to help narrow this down to the day when the bug started in RetroArch? (Aka would you help manually bisect this)

The question is essentially: would you work through the nightly PS Vita builds from March 10 till now caused the bug? Perhaps starting with April 10, for example, to see if it happened during the month of March 10 to April 10. 

Then once we know what month it took place in, splitting the month into two halves to further narrow the date, until there is a smaller window of time.

**Comment by evillurker** [comment]
> 
> 
> @evillurker would you be willing to help narrow this down to the day when the bug started in RetroArch? (Aka would you help manually bisect this)
> 
> The question is essentially: would you work through the nightly PS Vita builds from March 10 till now caused the bug? Perhaps starting with April 10, for example, to see if it happened during the month of March 10 to April 10.
> 
> Then once we know what month it took place in, splitting the month into two halves to further narrow the date, until there is a smaller window of time.

I don't have access to those builds and that would take me a long long time, i don't have that free time.
I'll just suffice with the no sounds then I guess

**Comment by markwkidd** [comment]
@evillurker if you have access to a RetroArch build from June 8, or a bit before then, that is one single test that might really help narrow it down for those of us who don't have access to your hardware.

Even this existing bug report is valuable, thank you either way.

**Comment by DerinQyu** [comment]
Hello vita slim user here.  Im using Jun 11 nightly (5d4069c) did a fresh install and there are no such bugs with the menu music or anything op says. Case on point I dont use DOSBox-SVN neither the mame2003plus cores. Things working fine with gambatte, pcsx rearmed, nestopia, mame2000 and gpsp though. Haven't touched stable 1.9.5 yet

June 12 nightly (3c2e3f1), did a fresh install too, is where the problems begin, sound is messed up after opening/closing quickmenu have to restart retroarch to fix it. Cores tested: pcsx-rearmed, gpsp, nestopia, gambatte, mame2000 and snes9x 2005.

----------------------------------------------------------------------------------------------------------------------------------

@evillurker mind sharing your March 10th nightly build with me please?

**Comment by markwkidd** [comment]
@twinaphex that narrows the range of commits down to a pretty narrow range between https://github.com/libretro/RetroArch/commit/5d4069cf8fcffb32e6a7dd139e9a5aa7e27c126d and https://github.com/libretro/RetroArch/commit/3c2e3f1ce9f0cd6fb302523da398a2d90f8677d5

**Comment by markwkidd** [comment]
Based on what @vudiq wrote in discord and the testing in this issue, this is likely the commit that introduced the problem https://github.com/libretro/RetroArch/commit/bd9c35d379bfc466cc2832e2c0ff7d1ed856da80

**Comment by evillurker** [comment]
Great job guys. I will test the build from june 11th and report if the issue with dosbox svn is occurring. The march cores can be obtained by simply googling it, if you still can't get it i will provide

**Comment by evillurker** [comment]
Update: I can confirm that the June 11th build does not crash DOSBOX-Svn and does not cause crackling sounds in mame 2003 plus or any other core.

**Comment by twinaphex** [comment]
Honestly I think you guys are conflating several issues.

Crackling sounds is not a bug, it's what happens when something falls below fullspeed. In this case, I would say Final Burn Neo (with whatever game you're using) is probably falling below fullspeed, hence the crackling.

Also, I'd say it's probably not wise for now to use the audio mixer on Vita at all. It's an additional resource burden on a system that already doesn't have a powerful enough CPU to speak of.

**Comment by evillurker** [comment]
> Honestly I think you guys are conflating several issues.
> 
> Crackling sounds is not a bug, it's what happens when something falls below fullspeed. In this case, I would say Final Burn Neo (with whatever game you're using) is probably falling below fullspeed, hence the crackling.
> 
> Also, I'd say it's probably not wise for now to use the audio mixer on Vita at all. It's an additional resource burden on a system that already doesn't have a powerful enough CPU to speak of.

If that was the case than it would happen on the march cores as well, which doesn't. Please don't speak of stuff you haven't even tested. Making assumptions based on your "opinion" just makes thing completely irrelevant.

**Comment by frangarcj** [comment]
could you send your retroarch.cfg?
I can't reproduce this problem

**Comment by evillurker** [comment]
Update: I tried the latest nightly build from june 27th. 
First of all they fixed the bug with dosbox svn being crashes.
Second, the bug when you scroll to the right with wallpaper'd playlists does not crash the vita anymore.

Lastly, if I disable menu sounds completely, this bug does not happen
To reproduce it, you have to enable menu sounds and then start a game and get into the menu, then resume the game.
At least I can use the latest version now but without any menu sounds.

**Comment by bielgiroto** [comment]
Vita Slim user here, 
I'm having issues with audio as well since 1.9.5 stable, 
The new 1.9.6 version didn't fixed much stuff.

With me it was and is very simillar, at first when I tried to turn on the bgm option it used to already have a broken sound, and whenever I tried to run a game from any core I had with it turned on, after resuming from opening the Retroarch menu this issue of having the game's audio completely distorded and crackled happened to me as well.

On 1.9.6 it fixed the audio crackling when first opening Retroarch, but I still can't play any game with this option turned on.
Oh and also, everytime I navigate through the playlists the song keeps poping, as if loading the rom images and bg somehow brokes a little the MP3 playback when I change playlists at the XMB menu.
